src/cache_parser.cpp

```diff
@@ -0,0 +1,67 @@
+#include "cache_parser.hpp"
+
+using json = nlohmann::json;
+
+void ParseCacheConfig(const char *filename, std::vector<CACHE_SET> &dest) {
+    json cache_conf;
+    std::ifstream file;
+    try {
+        file.open(filename);
+        file.exceptions(std::ifstream::eofbit | std::ifstream::failbit |
+                        std::ifstream::badbit);
+    } catch (std::exception const &e) {
+        std::cerr << e.what() << std::endl;
+    }
+    /*
+        Parse from json file on disk.
+        ref:
+       https://nlohmann.github.io/json/classnlohmann_1_1basic__json_af1efc2468e6022be6e35fc2944cabe4d.html
+    */
+    try {
+        file >> cache_conf;
+        file.close();
+    } catch (json::parse_error &e) {
+        std::cout << "message: " << e.what() << '\n'
+                  << "exception id: " << e.id << '\n'
+                  << "byte position of error: " << e.byte << std::endl;
+    }
+    auto _cache_array = cache_conf["content"];
+
+    for (json::iterator it = _cache_array.begin(); it < _cache_array.end();
+         it++) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer prefix ++/-- operators for non-primitive types.](https://app.codacy.com/app/jerry0715no14/Cache_sim/pullRequest?prid=4731053)